bin_sort/bin_sort.py

```diff
@@ -0,0 +1,84 @@
+"""
+Following is the implementation of bin sort algorithm.
+It will sort the input array with time complexity H(N + K)
```

Comment:
Well .. it's Theta(N + K) .. theta looks like H .. writing `O(N + K)` would be a mistake in the algorithm complexity